src/client/builtin_scene/materials/web_content_instanced.cpp

```diff
@@ -94,38 +94,56 @@ namespace builtin_scene::materials
     if (instancedMesh.instanceCount() <= 0)
       return;
 
-    // Update the render queues for opaque and transparent instances.
+    // a) Check isStructureDirty_ and update layeredInstances_ if needed
     instancedMesh.updateInstancesList();
 
     size_t meshIndicesCount = mesh.indices().size();
     CSSBorderDataTexture *borderDataTexture = getBorderDataTexture();
 
-    // Draw the transparent instances
-    RenderableInstancesList &instances = instancedMesh.getTransparentInstancesList();
-    if (instances.count() > 0)
-    {
-      WebGLVertexArrayScope vaoScope(glContext, instances.vao);
+    // b) Render layeredInstances_ in order (0-1-2-...)
+    instancedMesh.iterateLayers([&](RenderLayer layer, RenderableInstancesList &layerInstancesList)
+                                {
+      if (layerInstancesList.count() > 0)
+      {
+        // c) Switch RenderableInstancesList's vbo to current vao's vbo for this layer
+        WebGLVertexArrayScope vaoScope(glContext, layerInstancesList.vao);
+
+        // Set the base matrix, move the transparent objects +z 0.001
+        auto loc = glContext->getUniformLocation(program, "modelMatrix");
```

Comment:
`modelMatrix` is a shared uniform, you don't need to update this uniform for each batch of instances.